sim/gui.py

```diff
@@ -34,19 +34,132 @@ def __add__(self, other: object) -> Self:
         return type(self)(self.row + other.row, self.col + other.col)
 
 
-class GUIRenderer:
-    """Class for rendering a maze on screen"""
-    wall_thickness: int = 5
-    wall_color: str = 'red'
-    robot_main_color: str = 'blue'
-    robot_second_color: str = 'yellow'
-    goal_color: str = 'green'
-    heatmap_colors: list = ['blue', 'cyan', 'green', 'yellow', 'orange', 'red', 'brown']
+class GUIRenderer:  # pylint: disable=too-many-instance-attributes
+    """Class for rendering the GUI on screen"""
+
+    def __init__(self, sim: Simulator):
+        """Initialize GUI
+
+        Args:
+            sim (Simulator): The simulation to show.
+        """
+        pg.init()
+        initial_screen_size = (1280, 720)
+        screen = pg.display.set_mode(initial_screen_size, pg.RESIZABLE)
+        pg.display.set_caption('Micromouse')
+        self.screen = screen
+        self.wall_thickness: int = 5
+        self.wall_color: str = 'red'
+        self.robot_main_color: str = 'blue'
+        self.robot_second_color: str = 'yellow'
+        self.goal_color: str = 'green'
+        self.heatmap_colors: list = ['blue', 'cyan', 'green', 'yellow', 'orange', 'red', 'brown']
+        self.ui_manager = pygame_gui.UIManager((screen.get_width(), screen.get_height()))
+        self.start_button = pygame_gui.elements.UIButton(relative_rect=pg.Rect((0, 0), (0, 0)), text='Start', manager=self.ui_manager)
+        self.start_button.tool_tip_text = "Shortcut: 's'"
+        self.step_button = pygame_gui.elements.UIButton(relative_rect=pg.Rect((0, 0), (0, 0)), text='Step', manager=self.ui_manager)
+        self.step_button.tool_tip_text = "Shortcut: 'n'"
+        robots_list = ["Idle", "Random", "Left Wall Follower", "Right Wall Follower", "Flood Fill"]
+        self.robot_dropdown = pygame_gui.elements.UIDropDownMenu(robots_list, robots_list[0], pg.Rect((0, 0), (0, 0)), self.ui_manager)
+
+        self.sim_auto_step = False
+        self.sim = sim
+
+        screen_width, screen_height = initial_screen_size
+        self.tile_size = min(screen_width // (2 * self.sim.maze.width + 3), screen_height // (self.sim.maze.height + 4))
+        self.half_tile = self.tile_size // 2
+        self.full_maze_offset = Position(self.tile_size, 3 * self.tile_size)
+        self.full_maze_center = self.full_maze_offset + Position(self.sim.maze.width * self.half_tile, 0)
+        self.robot_maze_offset = self.full_maze_offset + Position((self.sim.maze.width + 1) * self.tile_size, 0)
+        self.robot_maze_center = self.robot_maze_offset + Position(self.sim.maze.width * self.half_tile, 0)
+
+    def run(self):
+        """Run the GUI
+        """
+        # self.configure_simulator()
+        self.scale()
+        self.main_loop()
+        pg.quit()
+
+    def get_selected_robot(self):
+        """Fetch the selected robot.
+
+        Returns:
+            Robot: The chosen robot
+        """
+        match (self.robot_dropdown.selected_option):
```

Comment:
No parentheses on match
```suggestion
        match self.robot_dropdown.selected_option:
```